go/vt/vtgate/vstream_manager_test.go

```diff
@@ -886,16 +886,37 @@ func TestResolveVStreamParams(t *testing.T) {
 		err:   "vgtid must have at least one value with a starting position",
 	}, {
 		input: &binlogdatapb.VGtid{
-			ShardGtids: []*binlogdatapb.ShardGtid{{}},
+			ShardGtids: []*binlogdatapb.ShardGtid{{
+				Keyspace: "TestVStream",
+			}},
```

Comment:
These test additions are unrelated to this PR, or no? I would think we would test the new behavior, which would mean NOT specifying the shards and now not getting an error?

Reply:
@mattlord I thought they were related to this PR.
Technically, I didn't add new test cases (input & output). I changed the expected output in the following existing two cases:

1. When the input is `ShardGtids: []*binlogdatapb.ShardGtid{{}},`
     - prev output: Returning an error
     - new output: Returning all shards in all keyspaces without any errors. 
     - Moving the case code to the special-case tables because output is too big. https://github.com/vitessio/vitess/pull/11909/files#diff-3e2894943e5628aa7d63298e06c321fe82602d18693b2d5e99541fafd2c01395R1021
2. When the input is `ShardGtids: []*binlogdatapb.ShardGtid{{Keyspace: "TestVStream"}},`
     - prev output: Returning an error
     - new output: Returning all shards in the TestVStream keyspace without any errors. 
     - https://github.com/vitessio/vitess/pull/11909/files#diff-3e2894943e5628aa7d63298e06c321fe82602d18693b2d5e99541fafd2c01395R893

Reply:
Here as well I think that comments noting what we're actually testing, how, and why are helpful. The future reader won't easily see comments here on the PR. And I don't know that the existing code is very poorly commented, but I want to do better moving forward. 🙂

Reply:
Added by [3b41d76](https://github.com/vitessio/vitess/pull/11909/commits/3b41d767bd0e6a9720bcaaa5ed9e3b7591540da3).